pkg/kv/kvserver/replica_metrics.go

```diff
@@ -86,43 +88,45 @@ func (r *Replica) Metrics(
 	lockTableMetrics := r.concMgr.LockTableMetrics()
 
 	return calcReplicaMetrics(ctx, calcReplicaMetricsInput{
```

Comment:
Consider creating the `calcReplicaMetricsInput` under the lock above, to dedup most of the variables.